DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -104,87 +146,328 @@ void FillBindingData(
+                    }
+                }
             }
         }
 
-        dmlOutputGraphEdges.resize(m_desc.bindPoints.outputs.size());
-        for( size_t i = 0; i < m_desc.bindPoints.outputs.size(); i++)
+        return constantDataTypes;
+}
+
+Dispatchable::Bindings GenerateInitialBindingsFromGraph(const DmlSerializedGraphDesc& serializedDesc,
+const std::unordered_map<std::string, DML_TENSOR_DATA_TYPE>& constantDataTypes)
+{
+    Dispatchable::Bindings local_bindings;
+
+    for (const auto& node : serializedDesc.Nodes)
+    {
+        if (std::holds_alternative<DmlSerializedGraphNodeConstantVariant>(node.Desc))
         {
-            if (m_desc.bindPoints.outputs[i].requiredBinding)
+            const auto& constantVariant = std::get<DmlSerializedGraphNodeConstantVariant>(node.Desc);
+            if (std::holds_alternative<ConstantName>(constantVariant))
             {
-                DML_OUTPUT_GRAPH_EDGE_DESC desc = {};
-                desc.GraphOutputIndex = gsl::narrow_cast<UINT>(i);
-                desc.FromNodeIndex = 0;
-                desc.FromNodeOutputIndex = gsl::narrow_cast<UINT>(i);
-                desc.Name = m_desc.bindPoints.outputs[i].name.c_str();
-                dmlOutputGraphEdges[i] = desc;
-                dmlOutputEdges.push_back({ DML_GRAPH_EDGE_TYPE_OUTPUT, &dmlOutputGraphEdges[i] });
+                Dispatchable::BindingSource bindingSource;
+                bindingSource.resource = nullptr;  
+                bindingSource.resourceDesc = nullptr;
+                bindingSource.elementOffset = 0;
+                bindingSource.elementCount = 0;  
+                bindingSource.elementSizeInBytes = GetElementSize(constantDataTypes.at(node.Name));
+                local_bindings[node.Name] = {bindingSource};
             }
         }
+    }
 
-        dmlGraphDesc.InputCount = static_cast<uint32_t>(dmlInputEdges.size());
-        dmlGraphDesc.InputEdges = dmlInputEdges.data();
-        dmlGraphDesc.InputEdgeCount = dmlGraphDesc.InputCount;
+    return local_bindings;
+}
 
-        dmlGraphDesc.OutputCount = static_cast<uint32_t>(dmlOutputEdges.size());
-        dmlGraphDesc.OutputEdges = dmlOutputEdges.data();
-        dmlGraphDesc.OutputEdgeCount = dmlGraphDesc.OutputCount;
+static std::vector<std::byte> LoadFileContents(const std::filesystem::path& filepath)
+{
+    std::ifstream file(filepath, std::ios::binary | std::ios::ate);
+    if (!file)
+    {
+        throw std::runtime_error("Could not open file: " + filepath.string());
+    }
 
-        dmlGraphDesc.IntermediateEdgeCount = 0;
-        dmlGraphDesc.IntermediateEdges = nullptr;
+    std::streamsize size = file.tellg();
+    file.seekg(0, std::ios::beg);
+    std::vector<std::byte> buffer(size);
+    
+    if (!file.read(reinterpret_cast<char*>(buffer.data()), size))
+    {
+        throw std::runtime_error("Could not read file: " + filepath.string());
+    }
+    return buffer;
+}
+
+void DmlDispatchable::CreateResourceFromConstantNode(
```

Comment:
>CreateResourceFromConstantNode [](http://example.com/codeflow?start=22&length=30)

This is a private class level method, so it can directly access all the private class level fields. No need to pass them explicitly like m_device, m_desc, m_initBindings.